packages/gatsby-dev-cli/src/local-npm-registry/verdaccio.js

```diff
@@ -0,0 +1,235 @@
+ *    install them in local site
+ */
+const adjustPackageJson = ({
+  monoRepoPackageJsonPath,
+  packageName,
+  versionPostFix,
+  packagesToPublish,
+  ignorePackageJSONChanges,
+}) => {
+  // we need to check if package depend on any other package to will be published and
+  // adjust version selector to point to dev version of package so local registry is used
+  // for dependencies.
+
+  const monorepoPKGjsonString = fs.readFileSync(
+    monoRepoPackageJsonPath,
+    `utf-8`
+  )
+  const monorepoPKGjson = JSON.parse(monorepoPKGjsonString)
+
+  monorepoPKGjson.version = `${monorepoPKGjson.version}-dev-${versionPostFix}`
+  packagesToPublish.forEach(packageThatWillBePublished => {
+    if (
+      monorepoPKGjson.dependencies &&
+      monorepoPKGjson.dependencies[packageThatWillBePublished]
+    ) {
+      // change to "gatsby-dev" dist tag
+      monorepoPKGjson.dependencies[packageThatWillBePublished] = `gatsby-dev`
+    }
+  })
+
+  const temporaryMonorepoPKGjsonString = JSON.stringify(monorepoPKGjson)
+
+  const unignorePackageJSONChanges = ignorePackageJSONChanges(packageName, [
+    monorepoPKGjsonString,
+    temporaryMonorepoPKGjsonString,
+  ])
+
+  // change version and dependency versions
+  fs.outputFileSync(monoRepoPackageJsonPath, temporaryMonorepoPKGjsonString)
+
+  return {
+    newPackageVersion: monorepoPKGjson.version,
+    unadjustPackageJson: () => {
+      // restore original package.json
+      fs.outputFileSync(monoRepoPackageJsonPath, monorepoPKGjsonString)
+      unignorePackageJSONChanges()
+    },
+  }
+}
+
+/**
+ * Anonymous publishing require dummy .npmrc
+ * See https://github.com/verdaccio/verdaccio/issues/212#issuecomment-308578500
+ * This is `npm publish` (as in linked comment) and `yarn publish` requirement.
+ * This is not verdaccio restriction.
+ */
+const createTemporaryNPMRC = ({ pathToPackage }) => {
+  const NPMRCPath = path.join(pathToPackage, `.npmrc`)
+  fs.outputFileSync(NPMRCPath, NPMRCContent)
+
+  return () => {
+    fs.removeSync(NPMRCPath)
+  }
+}
+
+const publishPackage = async ({
+  packageName,
+  packagesToPublish,
+  root,
+  versionPostFix,
+  ignorePackageJSONChanges,
+}) => {
+  const monoRepoPackageJsonPath = getMonorepoPackageJsonPath({
+    packageName,
+    root,
+  })
+
+  const { unadjustPackageJson, newPackageVersion } = adjustPackageJson({
+    monoRepoPackageJsonPath,
+    packageName,
+    root,
+    versionPostFix,
+    packagesToPublish,
+    ignorePackageJSONChanges,
+  })
+
+  const pathToPackage = path.dirname(monoRepoPackageJsonPath)
+
+  const uncreateTemporaryNPMRC = createTemporaryNPMRC({ pathToPackage })
+
+  // npm publish
+  const publishCmd = [
+    `npm`,
+    [`publish`, `--tag`, `gatsby-dev`, `--registry=${registryUrl}`],
+    {
+      cwd: pathToPackage,
+    },
+  ]
+
+  console.log(
+    `Publishing ${packageName}@${newPackageVersion} to local registry`
+  )
+  try {
+    await promisifiedSpawn(publishCmd)
```

Comment:
whenever I kill the process between publishing the npmrc & package.json aren't cleanup

Reply:
Right, I probably should use something like https://stackoverflow.com/questions/14031763/doing-a-cleanup-action-just-before-node-js-exits to do cleanup if it's in middle of publishing - dunno if it will work tho